common/src/test/java/io/druid/math/expr/EvalTest.java

```diff
@@ -19,114 +19,111 @@
-        bindings
-    ).longValue());
-    Assert.assertEquals(9223372036854775806L, evaluate(
-        "if(0, 9223372036854775807, 9223372036854775806)",
-        bindings
-    ).longValue());
+
+    Expr.ObjectBinding bindings = Parser.withMap(ImmutableMap.of("x", 9223372036854775807L));
+
+    Assert.assertEquals(9223372036854775807L, evalLong("x", bindings));
+    Assert.assertEquals(9223372036854775807L, evalLong("\"x\"", bindings));
+    Assert.assertEquals(92233720368547759L, evalLong("\"x\" / 100 + 1", bindings));
+
+    Assert.assertFalse(evalLong("9223372036854775807 && 0", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775807 && 9223372036854775806", bindings) > 0);
+
+    Assert.assertTrue(evalLong("9223372036854775807 || 0", bindings) > 0);
+    Assert.assertFalse(evalLong("-9223372036854775807 || -9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("-9223372036854775807 || 9223372036854775807", bindings) > 0);
+    Assert.assertFalse(evalLong("0 || 0", bindings) > 0);
+
+    Assert.assertTrue(evalLong("9223372036854775807 > 9223372036854775806", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775807 >= 9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775806 < 9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775807 <= 9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775807 == 9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("9223372036854775807 != 9223372036854775806", bindings) > 0);
+
+    Assert.assertEquals(9223372036854775807L, evalLong("9223372036854775806 + 1", bindings));
+    Assert.assertEquals(9223372036854775806L, evalLong("9223372036854775807 - 1", bindings));
+    Assert.assertEquals(9223372036854775806L, evalLong("4611686018427387903 * 2", bindings));
+    Assert.assertEquals(4611686018427387903L, evalLong("9223372036854775806 / 2", bindings));
+    Assert.assertEquals(7L, evalLong("9223372036854775807 % 9223372036854775800", bindings));
+    Assert.assertEquals(9223372030926249001L, evalLong("3037000499 ^ 2", bindings));
+    Assert.assertEquals(-9223372036854775807L, evalLong("-9223372036854775807", bindings));
+
+    Assert.assertTrue(evalLong("!-9223372036854775807", bindings) > 0);
+    Assert.assertTrue(evalLong("!0", bindings) > 0);
+    Assert.assertFalse(evalLong("!9223372036854775807", bindings) > 0);
+
+    Assert.assertEquals(3037000499L, evalLong("cast(sqrt(9223372036854775807), 'long')", bindings));
+    Assert.assertEquals(
+        9223372036854775807L, evalLong("if(9223372036854775807, 9223372036854775807, 9223372036854775806)", bindings)
+    );
+    Assert.assertEquals(
+        9223372036854775806L, evalLong("if(0, 9223372036854775807, 9223372036854775806)", bindings)
+    );
+
+    Assert.assertEquals(1271023381000L, evalLong("timestamp('2010-04-12T07:03:01+09:00')", bindings));
```

Comment:
Would be great to see tests for the rest of the functions in Function.java if possible, or at least for the ones that were added (cast, unix_timestamp, nvl)